lib/protocol/endpoint.js

```diff
@@ -29,11 +29,15 @@ module.exports = function(dependencies) {
 
     this._acquiredStreamSlots = 0;
     this._maximumStreamSlots = 0;
+    this._lastSuccessPingedTime = null;
+    this._pingedThreshold = (this.options.heartBeat || 60000) * 2;
+    this._heartBeatInterval = (this.options.heartBeat || 60000);
```

Comment:
Worth having `this._pingedThreshold` at `2.5` times? Otherwise we have a race condition going on if we fail a single heartbeat?

``` js
this._heartBeatInterval = (this.options.heartBeat || 60000);
this._pingedThreshold = this._heartBeatInterval * 2.5;
```

Seems more logical to me?